src/streaming-services/vrtnu-be/VrtnuBeParser.ts

```diff
@@ -0,0 +1,77 @@
+import { Item } from '../../models/Item';
+import { registerScrobbleParser } from '../common/common';
+import { ScrobbleParser } from '../common/ScrobbleController';
+import { VrtnuBeApi } from './VrtnuBeApi';
+
+class _VrtnuBeParser implements ScrobbleParser {
+	id: string;
+	videoId: string;
+	progress: number;
+	isPaused: boolean;
+
+	constructor() {
+		this.id = '';
+		this.videoId = '';
+		this.progress = 0.0;
+		this.isPaused = false;
+	}
+
+	parseProgress = (): number => {
+		let progress = 0.0;
+		const scrubber: HTMLElement | null = document.querySelector('.vjs-play-progress');
+
+		if (scrubber) {
+			progress = parseFloat(scrubber?.style.left);
+			this.progress = progress;
+		}
+
+		return progress;
+	};
+
+	parseItem = (): Item | undefined => {
+		const serviceId = 'vrtnu-be';
+		let showTitle: string | null = null;
+		let seasonOrYear: string | null = null;
+		let subTitle: string | undefined = undefined;
+		const matches = /\/a-z\/(.+)\/(.+)\/(.+)\//.exec(location.href);
+
+		if (matches) {
+			[, showTitle, seasonOrYear, subTitle] = matches;
+		}
+
+		const id = subTitle ?? '';
+		const title = showTitle?.split('-').join(' ') ?? '';
+		const episodeTitle = '';
+		const isSeason = subTitle?.lastIndexOf('s' + String(seasonOrYear) + 'a') ?? 0;
+		const episode =
+			isSeason > 0
+				? parseInt(subTitle?.substring(isSeason + ((seasonOrYear?.length ?? 0) + 2)) ?? '')
```

Comment:
I think this is very confusing. Perhaps you could type an example of the format as a comment. It could be easier for future debugging if things stops working, or the format is changes as well. Another approach could be a regular expression, but I guess either works.

Reply:
@ITSYV Looking back, you also used this logic for Disney Plus, which does make things a bit confusing. If you could use a regular expression for Disney Plus as well, like you did with the other services, I think it would be better. See [this](https://github.com/trakt-tools/universal-trakt-scrobbler/commit/81cef1ee778b802b76801a30189fcef5bb50b4e3#diff-d1fd825e3b0b628976a42f935a29148422a5aaa028b171432183015a39da4ddaR72-R75) vs [this](https://github.com/trakt-tools/universal-trakt-scrobbler/commit/436127ba79fc7189f22b10601e1287f1cc0f5a76#diff-382cf51bd6eb4ad35fbfdc29573c6df45c843378b66876bd59259461d22e3f0cR45-R54).

Reply:
"https://www.vrt.be/vrtnu/a-z/dertigers/3/dertigers-s3a1/#close"
VRT isn't using specific classes / id's for their elements and there is no specific logic in their buildup. So I can't scrape their page.

A regular expression could have also been working, but then I'm looping through possibly multiple matches, right?

Same for Disney+: I wasn't sure if there would be multiple "." or ":" after the season/episode info, as a title could have a ":" in it I guess. But a regex would work for sure :)

It's what you guys prefer, I can edit it if you want!

Reply:
> but then I'm looping through possibly multiple matches

I don't quite understand what you mean. Couldn't you easily modify the `/\/a-z\/(.+)\/(.+)\/(.+)\//` regex to directly get season/episode info like `/\/a-z\/(.+)\/(.+)\/(.+)-s(\d+)a(\d+)\//`, assuming it always ends in `-sXaY`?

Reply:
Well, if it's a movie, it doesn't end with sXaY ...
That's the reason for the extra check

Reply:
I'm guessing movies don't end with that, so maybe `/\/a-z\/(.+)\/(.+)\/(.+?)(-s(\d+)a(\d+))?\//`?

Reply:
Yeah, commented before I saw your reply. I think the regex I posted above would work. Then you could do:

[, showTitle, seasonOrYear, subTitle, _, season, episode]

Reply:
That could be a possibility yes! 
But I'm also using this as the ID for the movie. 

If I'm using that logic, I would end up with this, right? : 

```
const matches = /\/a-z\/(.+)\/(.+)\/(.+?)(-s(\d+)a(\d+))?\//.exec(location.href);
if (matches) {
	[, showTitle, seasonOrYear, subTitle, season, episode] = matches;
}
const id = subTitle ?? '' + season ? '-s' + season : '' + episode ? 'a' + episode : '';
```

Reply:
`(-s(\d+)a(\d+))` is 3 capture groups, so you'd end up with `[, showTitle, seasonOrYear, subTitle, seasonAndEpisode, season, episode]` where `seasonAndEpisode = '-s3a1'`, `season = 3` and `episode = 1`.

You could also put everything in another capture group: `((.+?)(-s(\d+)a(\d+)))`. This would lead to `[, showTitle, seasonOrYear, id, subTitle, seasonAndEpisode, season, episode]` and you'd get the ID directly.

Reply:
Learned something new again. Never used regex that advanced :)
I'll edit it to the last one 👍 

Do you like me to edit it for Disney+ too?

Reply:
If it's possible. I don't know the format for Disney+, maybe you can give us one example for that too?

Reply:
Disney+ formats the subtitles / episode titles as: "S1: afl. 6 One World, One People"
I used "." because it's language dependent

Another question about the previous, you said "seasonAndEpisode, season, episode" but season and episode have to be of type "number". Should I name them: seasonStr and episodeStr and afterwards do a parseInt? (or use the same names and do a parseInt later on?)

Reply:
Or can I use this when declaring: `let season: number | string | undefined = undefined;`
not sure if that's the way to go?

Reply:
> Should I name them: seasonStr and episodeStr and afterwards do a parseInt?

Yep, that's a lot better.

> Disney+ formats the subtitles / episode titles as: "S1: afl. 6 One World, One People"

Maybe something like `/(S(\d+).+?(\d+) )?(.+)/`? Then you'd have `[, seasonAndEpisode, seasonStr, episodeStr, subTitle]` and that would work for movies as well, assuming movies would be just 'One World, One People' without the beginning.

Reply:
> Yep, that's a lot better.
alright, I'll edit it to episodeStr 👍 

> Maybe something like `/(S(\d+).+?(\d+) )?(.+)/`? Then you'd have `[, seasonAndEpisode, seasonStr, episodeStr, subTitle]` and that would work for movies as well, assuming movies would be just 'One World, One People' without the beginning.
Movies don't even have a subtitle, so this won't even be a problem

Reply:
Perfect. Only one more thing, if the 'S' changes based on language, it's probably best to just do: `/(.+?(\d+).+?(\d+) )?(.+)/` instead.

Reply:
And we just have to hope that there's no language where the episode comes before the season. haha

Reply:
> Perfect. Only one more thing, if the 'S' changes based on language, it's probably best to just do: `/(.+?(\d+).+?(\d+) )?(.+)/` instead.

It could be that the "S" would change, not sure. In Dutch it's "seizoen", as "afl" is the abbreviation of "aflevering" = episode.